poolrpc/trader.proto

```diff
@@ -966,3 +987,45 @@ message StopDaemonRequest {
 
 message StopDaemonResponse {
 }
+
+message OfferSidecarRequest {
+    /*
+    The total channel capacity in satoshis. This will be used for the bid
+    order's amount and min channel/match size values.
+    */
+    uint64 channel_capacity_sat = 1;
+
+    /*
+    The number of satoshis that will be pushed to the recipient in the sidecar
+    channel resulting from the bid order submitted by the offering trader
+    (=provider). The pushed amount will be transferred from the provider's pool
+    account (=taker account) to the maker's account to reimburse them for the
+    balance they'll effectively be giving away. Makers can opt-out of being
+    matched with bid orders that request a push amount (currently only possible
+    as sidecar channels). The reimbursement between the recipient of the sidecar
+    channel and the provider (=taker) is _not_ part of the protocol and must
+    happen out of band. The sidecar protocol simply deducts all fees for the
+    sidecar channel (execution fee, lease premium, chain fees, push amount) from
+    the taker's trading account.
+    */
+    uint64 push_amount_sat = 2;
+}
+
+message SidecarTicket {
+    /*
+    The complete sidecar ticket in its string encoded form which is base64 URL
+    encoded, has a human readable prefix ('sidecar...') and a checksum built in.
+    The string encoded version will only be used on the trader side of the API.
+    All requests to the auctioneer expect the ticket to be in its raw, tlv
+    encoded byte form.
+    */
+    string ticket = 1;
+}
+
+message RegisterSidecarRequest {
+    /*
+    The sidecar ticket to register and add the node and channel funding
+    information to. The ticket must be in the state "offered".
+    */
+    string ticket = 1;
```

Comment:
Should this be using `SidecarTicket` itself vs the plain string version of it, as we might add some new fields over time.

Reply:
Wouldn't we end up adding those new fields to the ticket _itself_, as in the TLV encoded part? Or what new fields do you mean here?